src/uu/cut/src/whitespace_searcher.rs

```diff
@@ -0,0 +1,99 @@
+// This file is part of the uutils coreutils package.
+//
+// (c) Rolf Morel <rolfmorel@gmail.com>
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+// cSpell:ignore multispace
+
+use memchr::memchr2;
+
+pub struct WhitespaceSearcher<'a> {
+    haystack: &'a [u8],
+    position: usize,
+}
+
+impl<'a> WhitespaceSearcher<'a> {
+    pub fn new(haystack: &'a [u8]) -> WhitespaceSearcher<'a> {
+        WhitespaceSearcher {
+            haystack,
+            position: 0,
+        }
+    }
+}
+
+impl<'a> Iterator for WhitespaceSearcher<'a> {
```

Comment:
This implementation feels a bit backwards to me. Wouldn't it be easier to give ranges (or maybe even slices) of the fields instead of the whitespace? I.e.

```
first    second    third
^---^    ^----^    ^---^     // give these ranges
     ^--^      ^--^          // instead of these
```

I think that would lead to easier code in `cut_fields_whitespace`.

Reply:
Yes and no. I can come up with two reasons why I'd prefer iterating over delimiters, rather than fields.

1. `--only-delimited`/`-s` option. We need to know for each line whether there is at least one delimiter or not. With a peekable delimiter iterator, this is straightforward: `if delim_iter.peek().is_none() { ... }`
With the field iterator, on the other hand, this adds a bit of complexity as we need to test  if there are >= 2 field iterations.

2. consistency. The original implementation with a char delimiter also iterates over delimiters, not fields. See `Searcher`, `cut_fields_delimiter`, and `cut_fields`. I tried to follow the original logic as much as possible.

Reply:
That makes sense!